src/cc_catalog_airflow/dags/provider_api_scripts/raw_pixel.py

```diff
@@ -0,0 +1,187 @@
+        # inconsistent with the rescaled images
+        query_params = urlparse(img_url)
+        width = parse_qs(query_params.query).get("w", [])[0]
+        height = parse_qs(query_params.query).get("h", [])[0]
+        thumbnail = image.get("image_400", "")
+        return [img_url, width, height, thumbnail]
+    else:
+        logger.warning(f"Image not detected in URL: {foreign_url}")
+        return [None, None, None, None]
+
+
+def _get_title_owner(image):
+    title = image.get("image_title", "")
+    owner = image.get("artist_names", "")
+    owner = owner.replace("(Source)", "").strip()
+    return [title, owner]
+
+
+def _get_tags(image):
+    keywords = image.get("keywords_raw")
+    if keywords:
+        keyword_list = keywords.split(",")
+        keyword_list = [
+            word.strip()
+            for word in keyword_list
+            if word.strip()
+            not in ["cc0", "creative commons", "creative commons 0"]
+        ]
+        return keyword_list
+    else:
+        return []
+
+
+def _process_image_data(image):
+    # verify the license and extract the metadata
+    license = "cc0"
+    version = "1.0"
+
+    foreign_id, foreign_url = _get_foreign_id_url(image)
+    if not foreign_url:
+        return None
+    img_url, width, height, thumbnail = _get_image_properties(
+        image, foreign_url
+    )
+    if not img_url:
+        return None
+    title, owner = _get_title_owner(image)
+    tags = _get_tags(image)
+
+    # TODO:How to get license_url, creator_url, meta_data, source, watermarked?
+    return image_store.add_item(
+        foreign_landing_url=foreign_url,
+        image_url=img_url,
+        license_=license,
+        license_version=str(version),
+        foreign_identifier=str(foreign_id),
+        width=str(width) if width else None,
+        height=str(height) if height else None,
+        title=title if title else None,
+        raw_tags=json.dumps(tags, ensure_ascii=False) if bool(tags) else None,
+        creator=owner,
+        thumbnail_url=thumbnail,
+    )
+
+
+def _process_pages(total, result, page):
+    img_ctr = 0
+    is_valid = True
+
+    if not total:
+        total = 0
+        is_valid = False
+
+    while (img_ctr < total) and is_valid:
+        logger.info("Processing page: {}".format(page))
+
+        for img in result:
+            total_images = _process_image_data(img)
+            img_ctr = total_images if total_images else img_ctr
+
+        page += 1
+        total, result = _get_image_list(page)
+
+        if not result:
+            is_valid = False
+
+        if not total:
+            total = 0
+            is_valid = False
+
+    return img_ctr
+
+
+def main():
+    page = 1
+
+    logging.info("Begin: RawPixel API requests")
+
+    total, result = _get_image_list(page)
+
+    img_ctr = _process_pages(total, result, page=page)
+
+    logging.info("Total images: {}".format(img_ctr))
+
+    image_store.commit()
+
+    logging.info("Terminated!")
```

Comment:
please change `logging` to `logger` in these calls.

Reply:
I'm sorry. I thought we should use logger for modules only.